node-manager/src/ldkstorage.rs

```diff
@@ -0,0 +1,135 @@
+use std::io;
+use std::io::Cursor;
+use std::ops::Deref;
+use std::sync::Arc;
+
+use bitcoin::BlockHash;
+use lightning::chain::channelmonitor::ChannelMonitor;
+use lightning::chain::keysinterface::{KeysInterface, Sign};
+use lightning::routing::scoring::{ProbabilisticScorer, ProbabilisticScoringParameters};
+use lightning::util::persist::KVStorePersister;
+use lightning::util::ser::{ReadableArgs, Writeable};
+use log::error;
+
+use crate::localstorage::MutinyBrowserStorage;
+use crate::logging::MutinyLogger;
+use crate::node::NetworkGraph;
+
+pub struct MutinyNodePersister {
+    node_id: String,
+    storage: MutinyBrowserStorage,
+}
+
+impl MutinyNodePersister {
+    fn get_key(&self, key: &str) -> String {
+        format!("{}_{}", key, self.node_id)
+    }
+
+    fn read_value(&self, _key: &str) -> Result<Vec<u8>, io::Error> {
+        let key = self.get_key(_key);
+        self.storage.get(key).map_err(io::Error::other)
+    }
+
+    pub fn persist_network_graph(&self, network_graph: &NetworkGraph) -> io::Result<()> {
+        self.persist("network", network_graph)
+    }
+
+    pub fn read_network_graph(
+        &self,
+        genesis_hash: BlockHash,
+        logger: Arc<MutinyLogger>,
+    ) -> NetworkGraph {
+        let (already_init, kv_value) = match self.read_value("network") {
```

Comment:
We could probably get rid of this `already_init` stuff completely TBH. I forgot why I did that before. But instead of returning an empty vec and later deciding when to initialize from scratch or read the key values, maybe we can put it all in this `match self.read_value` scope.